Assets/Scripts/EnemyController.cs

```diff
@@ -49,24 +69,21 @@ void FixedUpdate()
     public void ChangeHealth(int healthLost)
     {
         currentHealth += healthLost;
-        if(currentHealth <= 0)
-        {
-            Die();
-        }
+
+        if (currentHealth <= 0) { Die(); }
     }
 
     void Die()
     {
+        if (player != null)
+        {
+            player.enemyHash.Remove(thisTransform);
+            inHash = false;
+        }
         Object.Destroy(this.gameObject);
     }
 
-    bool IsWithinCamera()
-    {
-        if (GetComponent<Renderer>().isVisible)
-        {
-            return true;
-        }else return false;
-    }
+    bool IsWithinCamera() => (GetComponent<Renderer>().isVisible) ? true : false;
```

Comment:
I think it's better to omit `? true : false`.

Reply:
Do you mean that notation is unnecessary to perform this or use the regular notation for an if statement?

Reply:
It's unnecessary to use `? true : false`. I think that `bool IsWithinCamera() => spriteRenderer.isVisible;` should be used instead.